driver-sync/src/test/functional/com/mongodb/client/ChangeStreamProseTest.java

```diff
@@ -0,0 +1,429 @@
+/*
+ * Copyright 2008-present MongoDB, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.mongodb.client;
+
+import com.mongodb.MongoChangeStreamException;
+import com.mongodb.MongoCommandException;
+import com.mongodb.MongoException;
+import com.mongodb.MongoQueryException;
+import com.mongodb.client.internal.MongoChangeStreamCursorImpl;
+import com.mongodb.client.model.Aggregates;
+import com.mongodb.client.model.changestream.ChangeStreamDocument;
+import com.mongodb.operation.AggregateResponseBatchCursor;
+import org.bson.BsonArray;
+import org.bson.BsonDocument;
+import org.bson.BsonInt32;
+import org.bson.BsonString;
+import org.bson.Document;
+import org.junit.Before;
+import org.junit.Test;
+
+import java.lang.reflect.Field;
+
+import static com.mongodb.ClusterFixture.isDiscoverableReplicaSet;
+import static com.mongodb.ClusterFixture.serverVersionAtLeast;
+import static com.mongodb.ClusterFixture.serverVersionLessThan;
+import static java.util.Arrays.asList;
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertNotEquals;
+import static org.junit.Assert.assertNotNull;
+import static org.junit.Assert.assertNull;
+import static org.junit.Assert.assertTrue;
+import static org.junit.Assume.assumeTrue;
+
+// See https://github.com/mongodb/specifications/tree/master/source/change-streams/tests/README.rst#prose-tests
+public class ChangeStreamProseTest extends DatabaseTestCase {
```

Comment:
Good spot - there will need to be an async ChangeStreamProseTest

Reply:
Whoops! That clearly slipped by me. I will work on that.

Reply:
Async change streams do not implement a cursor that iterates over single documents. There is a `batchCursor()` method that returns an entire batch, but there is no equivalent to `MongoCursor` on the sync side where a single document is returned. Because of this, a `getResumeToken()` method cannot be implemented.

Reply:
I'm not sure about differences in `sync` and `async` code in Java, but even if we cannot implement provided prose tests.. , we need to add additional tests which will cover all cases which we support for async code